### PR TITLE
Repo Analytics

### LINKED ISSUE #2108: Track Github metrics over time
We currently have little insight into how well we are maintaining our Github page.

It would be useful to have some way of tracking some metrics over time, to see whether we are improving / getting worse.

Some things we could track:

* Issue resolution time (how long does it take before we close an issue) (e.g., as in isitmaintained.com)
* Number of open issues (isitmaintained.com)
* Issue response time (how long does it take before we reply to an issue)

As a motivation: when querying isitmaintained.com on April 2022, we get the following scores for "issue resolution time":
* Flax: 21d
* JAX: 4d
* Tensorflow 8d
* Pytorch: 6d

Clearly we can improve here as Flax!

### PR DESCRIPTION
**Pull request by cgarciae**
# What does this PR do?
Adds a script to fetch and visualize metrics of the Flax repo from Github.

Closes #2108

### CODE REVIEW

File: .github/analytics/issue_activity_since_date.gql [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
+  repository(owner: "_REPO_OWNER_", name: "_REPO_NAME_") {
```
Comment (marcvanzee) [inline .github/analytics/issue_activity_since_date.gql:2]: Could you add some comments to these files explaining what exactly they are querying?

File: .github/analytics/get_repo_metrics.py [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+  )
+  issues.get()
+
+  df_issues = pd.DataFrame(list(_get_issues_features(issues.raw_data)))
+  df_issues["issue_response_time"] = df_issues["time_labeled_or_converted"] - df_issues["created_at"]
+  df_issues["issue_resolution_time"] = df_issues["time_issue_closed"] - df_issues["created_at"]
+
+  df_issues = _rolling_window(df_issues, _process_issues)
+
+  # TODO: Analyze PR data
```
Comment (marcvanzee) [inline .github/analytics/get_repo_metrics.py:291]: This doesn't work yet? Or you just haven't looked at it yet?

**Review by marcvanzee — CHANGES_REQUESTED** [review]
Very cool! So it seems you can also very easily apply this to other repos, right? I think it would be cool to chat about this during the next sync with HuggingFace and ask them whether they are tracking any metrics.

Also, it may be useful for you to glance over the [Google Python Styleguide](https://google.github.io/styleguide/pyguide.html), which we try to follow as much as possible in our project (although we aren't perfect).

**Comment by cgarciae** [comment]
Thanks @marcvanzee!

> So it seems you can also very easily apply this to other repos, right?

Yes, I believe if you just change `google` and `flax` on `main` to any other user + repo it should work.

> Also, it may be useful for you to glance over the [Google Python Styleguide](https://google.github.io/styleguide/pyguide.html), which we try to follow as much as possible in our project (although we aren't perfect).

Yeah, sorry about that, most this code is ported from another project, I will fix it to match this project and remove references to non-existing stuff to avoid confusion.

**Comment by cgarciae** [comment]
Added 2 metrics for the PRs and modified the ported code a bit to match our style.

**Comment by cgarciae** [comment]
Many of the comments were about the ported code, I added a comment noting the source.
Also added comments explaining a bit what each query does.
Finally, last commit adds 2 metrics for PRs.

**Review by marcvanzee — APPROVED** [review]
Looks good now, thanks a lot!

**Pull request merged**